streams/runners/storm/actions/src/main/java/com/hortonworks/streamline/streams/actions/storm/topology/StormTopologyActionsImpl.java

```diff
@@ -765,6 +785,65 @@ private ShellProcessResult executeShellProcess (List<String> commands) throws  E
         return new ShellProcessResult(exitValue, stdout);
     }
 
+    private ShellProcessResult waitTestRunProcess(Process process, long topologyRunHistoryId) throws IOException {
+        forceKillRequests.put(topologyRunHistoryId, false);
+
+        LOG.info("Waiting for test run for history {} to be finished...", topologyRunHistoryId);
+
+        StringBuilder sb = new StringBuilder();
+        for (int idx = 0 ; ; idx++) {
+            if (!process.isAlive()) {
+                break;
+            }
+
+            // read stdout if available
+            sb.append(readProcessStream(process.getInputStream()));
+
+            if (forceKillRequests.getOrDefault(topologyRunHistoryId, false)) {
+                LOG.info("Received force kill for test run {} - destroying process...", topologyRunHistoryId);
+                process.destroyForcibly();
+            }
+
+            // leave a log for each 10 secs...
+            if ((idx + 1) % 100 == 0) {
+                LOG.info("Still waiting for test run for history {} to be finished...", topologyRunHistoryId);
+            }
+
+            try {
+                process.waitFor(100, TimeUnit.MILLISECONDS);
```

Comment:
maybe exit only when waitFor returns true? not sure if wait hasn't yet returned true it will cause zombies.

Reply:
This loop doesn't terminate unless process has exit code: isAlive() checks whether exit code is available or not.